File: spring_java/src/main/java/dev/pollito/spring_java/sakila/auth/domain/port/in/AuthUseCases.java
package dev.pollito.spring_java.sakila.auth.domain.port.in;

import dev.pollito.spring_java.config.security.userdetails.SakilaUserDetails;

public interface AuthUseCases {
  String authenticate(String username, String password);

  SakilaUserDetails getCurrentUser();
}

File: spring_kotlin/src/main/kotlin/dev/pollito/spring_kotlin/sakila/auth/domain/port/in/AuthUseCases.kt
package dev.pollito.spring_kotlin.sakila.auth.domain.port.`in`

import dev.pollito.spring_kotlin.config.security.userdetails.SakilaUserDetails

interface AuthUseCases {
  fun authenticate(username: String, password: String): String

  fun getCurrentUser(): SakilaUserDetails
}
